src/initialize.cc

```diff
@@ -374,6 +377,7 @@ std::shared_ptr<void(int)> HANDLER_NAME(                             \
 SIGNAL_HANDLER(SIGSEGV, SIGSEGVHandler, true);
 SIGNAL_HANDLER(SIGFPE, SIGFPEHandler, false);
 SIGNAL_HANDLER(SIGBUS, SIGBUSHandler, false);
+SIGNAL_HANDLER(SIGPIPE, SIGPIPEHandler, false);
```

Comment:
What's the point in handling `SIGPIPE` in mxnet? Consider `foo | bar` where `foo` uses mxnet. Once `bar` dies, `foo` should be terminated, but with this PR `foo` will keep running.

Reply:
it might also be that mxnet is saving to a file while the inode was moved.

Reply:
But would that justify breaking all users that pipe output of a program which opened `libmxnet.so`?

Reply:
what do you mean by breaking?

Reply:
In the example you provided, I think users can disable the signal handler using the build option if they intend to link it to `foo`.

Reply:
The PR as it currently stands breaks `python3 script.py | head` in that script.py will not be terminated after 10 lines of output in case it imported mxnet.